core/wiki/macros/CSS.tid

```diff
@@ -1,43 +1,60 @@
 title: $:/core/macros/CSS
 tags: $:/tags/Macro
 
-\procedure actions-compile-palette-filtered(tempPalette,outputTitle)
+\procedure actions-compile-palette-filtered(tempPalette,outputTitle,scheme)
 <!-- Note the join, needed to cope with palette entries containing spaces -->
-\function colour(name) [<tempPalette>getindex<name>] :else[[$:/config/DefaultColourMappings/]addsuffix<name>get[text]] :map[subfilter<currentTiddler>join[ ]]
+\function colour(name) [<name>addsuffix[@]addsuffix<scheme>] :map[<tempPalette>getindex<currentTiddler>] +[!match[]] :else[<tempPalette>getindex<name>] :else[[$:/config/DefaultColourMappings/]addsuffix<name>addsuffix[@]addsuffix<scheme>get[text]] :else[[$:/config/DefaultColourMappings/]addsuffix<name>get[text]] :map[subfilter<currentTiddler>join[ ]]
+\function color(name) [function[colour],<name>]
 <!-- Make the colour function use the input palette -->
 <$list filter="[<tempPalette>indexes[]sort[]]" variable="colour-name">
-	<$let colour-filter={{{ [<tempPalette>getindex<colour-name>] }}}>
+	<$let filter-text={{{ [<colour-name>addsuffix[@]addsuffix<scheme>] :map[<tempPalette>getindex<currentTiddler>] +[!match[]] :else[<tempPalette>getindex<colour-name>] :else[[$:/config/DefaultColourMappings/]addsuffix<colour-name>addsuffix[@]addsuffix<scheme>get[text]] :else[[$:/config/DefaultColourMappings/]addsuffix<colour-name>get[text]] }}}>
 		<!-- Note the join, needed to cope with palette entries containing spaces -->
-		<$action-setfield $tiddler=<<outputTitle>> $index=<<colour-name>> $value={{{ [subfilter<colour-filter>join[ ]] }}}/>
+		<$action-setfield $tiddler=<<outputTitle>> $index=<<colour-name>> $value={{{ [subfilter<filter-text>join[ ]] }}}/>
 	</$let>
 </$list>
 \end actions-compile-palette-filtered
 
-\procedure actions-compile-palette-wikified(tempPalette,outputTitle)
+\procedure actions-compile-palette-wikified(tempPalette,outputTitle,scheme)
 <!-- Define our own deeply backwards compatible local versions of the colour macro -->
 \define colour(name)
```

Comment:
There are 2 `\define` inside the procedure. Is this intentional?

Reply:
Yes. They are nested definitions. A macro is being used here for backwards compatibility.